src/managed/OpenLiveWriter.BlogClient/Clients/WordPressClient.cs

```diff
@@ -53,5 +54,49 @@ protected override void ConfigureClientOptions(BlogClientOptions clientOptions)
             clientOptions.SupportsAuthor = true;
         }
 
+        public override BlogInfo[] GetUsersBlogs()
+        {
+            TransientCredentials tc = Login();
+            return GetWPUsersBlogs(tc.Username, tc.Password);
+        }
+        private BlogInfo[] GetWPUsersBlogs(string username, string password)
+        {
+            // call method
+            XmlNode result = CallMethod("wp.getUsersBlogs",
+                new XmlRpcString(username),
+                new XmlRpcString(password, true));
+
+            try
+            {
+                // parse results
+                ArrayList blogs = new ArrayList();
+                XmlNodeList dataValues = result.SelectNodes("array/data");
+                foreach (XmlNode dataValue in dataValues)
+                {
+                    XmlNodeList blogNodes = dataValue.SelectNodes("value/struct");
+                    foreach (XmlNode blogNode in blogNodes)
+                    {
+                        // get node values
+                        XmlNode idNode = blogNode.SelectSingleNode("member[name='blogid']/value");
+                        XmlNode nameNode = blogNode.SelectSingleNode("member[name='blogName']/value");
+                        XmlNode urlNode = blogNode.SelectSingleNode("member[name='url']/value");
+
+                        // add to our list of blogs
+                        blogs.Add(new BlogInfo(idNode.InnerText, HttpUtility.HtmlDecode(NodeToText(nameNode)), urlNode.InnerText));
+                    }
+
+                    // return list of blogs
+                    return (BlogInfo[])blogs.ToArray(typeof(BlogInfo));
```

Comment:
I didn't catch that, thank you. I'm going to see what happens when the return includes multiple blogs now... Could have been a happy accident as far as the first in the array was the root blog, sub domain www, instead of my other blog, ptg.